File: api/routers/schedule_tradeoffs.py
"""Persisted MCQ answers for schedule intake (per day)."""

from __future__ import annotations

from datetime import date

from fastapi import APIRouter, Body, HTTPException, Query

from api.services.schedule_tradeoff_store import get_answers_for_day, put_answers_for_day

router = APIRouter(prefix="/schedule-tradeoffs", tags=["schedule-tradeoffs"])


@router.get("")
def get_schedule_tradeoffs(day: date = Query(..., description="Calendar day")) -> dict:
    return {"date": day.isoformat(), "answers": get_answers_for_day(day)}


@router.put("")
def put_schedule_tradeoffs(
    day: date = Query(..., description="Calendar day"),
    body: dict = Body(...),
) -> dict:
    try:
        merged = put_answers_for_day(day, body)
    except ValueError as e:
        raise HTTPException(status_code=400, detail=str(e)) from e
    return {"date": day.isoformat(), "answers": merged}

File: web/src/api/scheduleTradeoffs.ts
export async function getScheduleTradeoffs(day: string): Promise<{ date: string; answers: Record<string, string> }> {
  const r = await fetch(`/api/schedule-tradeoffs?day=${encodeURIComponent(day)}`);
  if (!r.ok) throw new Error(await r.text());
  return r.json() as Promise<{ date: string; answers: Record<string, string> }>;
}

export async function putScheduleTradeoffs(
  day: string,
  body: Record<string, string>,
): Promise<{ date: string; answers: Record<string, string> }> {
  const r = await fetch(`/api/schedule-tradeoffs?day=${encodeURIComponent(day)}`, {
    method: "PUT",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(body),
  });
  if (!r.ok) throw new Error(await r.text());
  return r.json() as Promise<{ date: string; answers: Record<string, string> }>;
}
